### DISCUSSION TITLE
Flask socketio (using gevent): are room messages emitted in order ?

### DISCUSSION THREAD

**Discussion by cty512** [miguelgrinberg/Flask-SocketIO#1833] [Q&A]
Hi there,

Good day.
I have a quick query:

we have Flask-Socketio (using gevent) as server forwarding room message like below:

simplified from example [app_namespace.py:](https://github.com/miguelgrinberg/Flask-SocketIO/blob/94a28590e211784e9365ac21185fd617bd1a0a9f/example/app_namespace.py)

```python
    def on_my_room_event(self, message): 
        emit('my_response', {'data': message['data']}, room=message['room'])
```

if
client send '1' to room1
client send '2' to room1

will room message be emitted by server in order ?
or in extreme case, will gevent schedule **on_my_room_event()** to emit message '2' prior to '1' ?

**Comment by miguelgrinberg** [comment]
This depends on implementation details of the gevent scheduler, it is not a question that I can answer. If you want the messages from a client to be processed in strict order, pass `async_handler=False` when you create your server, and then the server will invoke handlers for each client back to back instead of in parallel.

**Reply by cty512** [reply]
Thank you so much for your advice.
